symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/health/HealthService.java

```diff
@@ -0,0 +1,95 @@
+package com.symphony.bdk.core.service.health;
+
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.gen.api.SignalsApi;
+import com.symphony.bdk.gen.api.SystemApi;
+import com.symphony.bdk.gen.api.model.AgentInfo;
+import com.symphony.bdk.gen.api.model.V2HealthCheckResponse;
+import com.symphony.bdk.gen.api.model.V3Health;
+import com.symphony.bdk.http.api.ApiException;
+import com.symphony.bdk.http.api.ApiRuntimeException;
+
+import org.apiguardian.api.API;
+
+/**
+ * Service class for checking health of the Agent server.
+ */
+@API(status = API.Status.STABLE)
+public class HealthService {
+
+  private final SystemApi systemApi;
+  private final SignalsApi signalsApi;
+  private final AuthSession authSession;
+
+  public HealthService(SystemApi systemApi, SignalsApi signalsApi, AuthSession authSession) {
+    this.systemApi = systemApi;
+    this.signalsApi = signalsApi;
+    this.authSession = authSession;
+  }
+
+  /**
+   * Returns the connectivity status of your Agent server.
+   * If your Agent server is started and running properly, the status value will be UP.
+   * Available on Agent 2.57.0 and above.
+   *
+   * @return {@link V3Health} the connectivity status of your Agent server.
+   * @see <a href="https://developers.symphony.com/restapi/reference#health-check-v3">Health Check v3</a>
+   */
+  public V3Health v3HealthCheck() {
+    try {
+      return systemApi.v3Health();
+    } catch (ApiException e) {
+      throw new ApiRuntimeException(e);
+    }
+  }
+
+  /**
+   * Returns the connectivity status of the Agent services as well as users connectivity.
+   * Available on Agent 2.57.0 and above.
+   *
+   * @return {@link V3Health} the connectivity status of the Agent services as well as users connectivity.
+   * @see <a href="https://developers.symphony.com/restapi/reference#health-check-extended-v3">Healt Check Extended v3</a>
+   */
+  public V3Health v3ExtendedHealthCheck() {
```

Comment:
do we want to expose all of them? like do we need the v2 one for instance?

Reply:
I think the v2 and the extended v3 is mandatory because the v3 healthcheck is not available on all environment. Not sure if we need the default v3 health check, it only returned the status of the agent server.